src/vm/dllimport.cpp

```diff
@@ -4194,9 +4194,13 @@ static void CreateNDirectStubWorker(StubState*         pss,
     bool fHasCopyCtorArgs = false;
     bool fStubNeedsCOM = SF_IsCOMStub(dwStubFlags);
 
+#if defined(_TARGET_X86_) || defined(_TARGET_ARM_)
```

Comment:
Should this be _TARGET_X86 only? I suppose ARM JIT could handle Struct Foo() correctly. The ABI difference should be handled by JIT.